File: 1051-height-checker/1051-height-checker.py
class Solution:
    def heightChecker(self, heights: List[int]) -> int:
        n, ans = max(heights), 0
        count = [0] * (n + 1)
        
        for height in heights:
            count[height] += 1
        
        i = 1
        for height in heights:
            while count[i] == 0:
                i += 1
            
            if height != i:
                ans += 1
            count[i] -= 1
                
        return ans

File: 1051-height-checker/1051-height-checker.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        vector<int> sortedHeights = heights;
        sort(sortedHeights.begin(), sortedHeights.end());
        
        int ans = 0;
        for (int i = 0; i < heights.size(); ++i)
            if (sortedHeights[i] != heights[i])
                ++ans;
        
        return ans;
    }
};